Sources/UIKit+Diff.swift

```diff
@@ -88,29 +66,89 @@ public extension UITableView {
         apply(diff: diff.diff, insertRowAnimation: insert, deleteRowAnimation: delete)
         updateVisible(rows: diff.updates, updateVisibleRow)
     }
+
+    public func applyWithoutBatch(diff: Set<DiffStep<Int>>, section: Int,
+                                  insertRowAnimation: UITableViewRowAnimation = .automatic,
+                                  deleteRowAnimation: UITableViewRowAnimation = .automatic) {
+        var insertionIndexPaths: [IndexPath] = []
+        var deletionIndexPaths: [IndexPath] = []
+        for step in diff {
+            switch step {
+            case .insert(let index):
+                insertionIndexPaths.append(IndexPath(row: index, section: section))
+            case .delete(let index):
+                deletionIndexPaths.append(IndexPath(row: index, section: section))
+            case .move(let from, let to):
+                moveRow(at: IndexPath(row: from, section: section),
+                        to: IndexPath(row: to, section: section))
+            }
+        }
+        insertRows(at: insertionIndexPaths, with: insertRowAnimation)
+        deleteRows(at: deletionIndexPaths, with: deleteRowAnimation)
+    }
+
+    public func applyWithoutBatch(diff: Set<DiffStep<IndexPath>>,
+                                  insertRowAnimation: UITableViewRowAnimation = .automatic,
+                                  deleteRowAnimation: UITableViewRowAnimation = .automatic) {
+        var insertionIndexPaths: [IndexPath] = []
+        var deletionIndexPaths: [IndexPath] = []
+        for step in diff {
+            switch step {
+            case .insert(let index):
+                insertionIndexPaths.append(index)
+            case .delete(let index):
+                deletionIndexPaths.append(index)
+            case .move(let from, let to):
+                moveRow(at: from, to: to)
+            }
+        }
+        insertRows(at: insertionIndexPaths, with: insertRowAnimation)
+        deleteRows(at: deletionIndexPaths, with: deleteRowAnimation)
+    }
 }
 
 public extension UICollectionView {
+    public func applyWithoutBatch(diff: Set<DiffStep<Int>>, section: Int) {
```

Comment:
decided to not have a default section because it seems that if you were using this you would want to choose the section more than in the batch case. Maybe not though. Maybe the default parameter is a bad choice anyways.